README.md

```diff
@@ -1,73 +1,320 @@
+ 
+| Parameter            | Description                                                                                       | Type        | Required | Default                 | Libreswan Equivalent |
+|----------------------|---------------------------------------------------------------------------------------------------|:-----------:|:--------:|-------------------------|:--------------------:|
+| host                 | A host/IP to use for setting up a vpn connection. Only specified when overriding the name of hosts used by Ansible for SSH. | str | no   | - | left/right           |
+| public_key_src       | A path to a file on the controller host containing the public key used by this host for public key authentication without certificates.     | str    | no       | See [public_key fields](#public_key)| leftrsasigkey/rightrsasigkey |
+| public_key_content   | Contains the public key used by this host for public key authentication without certificates.                        | str         | no       | -                       | leftrsasigkey/rightrsasigkey |
+| cert_name            | Defines the certificate nickname of this host's certificate in the NSS database.                  | str         | no       | -                       | leftcert/rightcert   |
+| external             | Indicates that the associated host is external to the inventory (e.g. not managed by our inventory).    | bool        | no       | False                   | -                    |
+| subnets              | Defines a list of the subnets that should be available via the VPN connection.                    | list        | no       | -                       | leftsubnets/rightsubnets |
+
+#### public_key
+
+`public_key_src` specifies a path to a file on the controller host containing the public key used by this host for public key authentication without certificates. Otherwise, the user can directly specify the public key for this host by populating `public_key_content`. `public_key_content` can also accept a CKAID or nickname for a public key in the NSS database.
+
+Note that `public_key_src` and `public_key_content` may also be specified as host variables in the Ansible `all.hosts` section. The variable names in this case will be `vpn_public_key_src` and `vpn_public_key_content`. See [this example](#defining-public-keys-at-different-levels).
+
+**To be added later**: if neither `public_key_src` nor `public_key_content` is populated, the role will generate key pairs for each host.
+
+### Handling special cases with host names
+
+There are two ways to handle the special case when the names of the hosts in the Ansible inventory are not FQDN hostnames that you can use directly (e.g. they may be host aliases like "bastion_east"):  
+
+1. **Use the Ansible `hosts` configuration**
 
 ```yaml
-template_foo: "oof"
-template_bar: "baz"
+all:
+  hosts:
+    bastion_east:
+      ansible_host: bastion1.example.com
+      vpn_hostid: '@bastion-east'
```

Comment:
What is this `vpn_hostid` parameter? I don't see it mentioned anywhere else in the document.

Reply:
`vpn_hostid` is mentioned here [https://github.com/mprovenc/vpn#handling-special-cases-with-host-names](https://github.com/mprovenc/vpn#handling-special-cases-with-host-names). I agree that it could be helpful to have a more obvious description somewhere in the document. It doesn't really fit into any of the existing sections, so maybe a new section for host variables.

Reply:
"is mentioned here https://github.com/mprovenc/vpn#handling-special-cases-with-host-names" - but this is the very place we are commenting on! I mean, the parameter is not documented. It is not said what the parameter is, what it is supposed to do.

Reply:
Perhaps we should just omit `vpn_hostid` for now as it is a libreswan specific field and I think we can generate it.

Reply:
Well, my understanding of the `vpn_hostid` field was that it was an alternate way for users to refer to the hosts in the inventory so that (1) the libreswan `leftid` and `rightid` field could be populated with something other than the IP address/FQDN and (2) the connection could be named with something other than a long FQDN (e.g. `bastion1-to-bastion2` instead of `bastion1.reallylongfqdn.example.com-to-bastion2.reallylongfqdn.example.com`). Of course we can generate this value, but my question is how would we guarantee that it is something that the user can recognize when they see the auto-generated connection name? Or maybe for the MVP we just use the corresponding `ansible_host` value for the connection name regardless of how long the value is?

Reply:
> Well, my understanding of the `vpn_hostid` field was that it was an alternate way for users to refer to the hosts in the inventory so that (1) the libreswan `leftid` and `rightid` field could be populated with something other than the IP address/FQDN and (2) the connection could be named with something other than a long FQDN (e.g. `bastion1-to-bastion2` instead of `bastion1.reallylongfqdn.example.com-to-bastion2.reallylongfqdn.example.com`).

Correct.  But that is a libreswan specific thing, needing some sort of identifier for the left/right side of the tunnel, and something that is printed to users for monitoring - I don't think there is anything comparable in wireguard, openvpn, etc.

> Of course we can generate this value, but my question is how would we guarantee that it is something that the user can recognize when they see the auto-generated connection name?

I don't think it is important for MVP.

> Or maybe for the MVP we just use the corresponding `ansible_host` value for the connection name regardless of how long the value is?

For MVP I think we can use what you have in https://github.com/linux-system-roles/vpn/pull/3/files#diff-faa4f99711190a92b1f43259cdce080b2e2e84636823f6f099c2c62ca03d5cc9R7-R8

Reply:
As a side note, I was going through the README again and I saw that there is an option to to override the role-generated connection name as described here https://github.com/mprovenc/vpn#name. I had forgotten that we planned on providing that option, but if we keep it there, that would potentially address the issue of long connection names.